### PR TITLE
Optimize UNet by introducing dilated convolutions and skip connections

### PR DESCRIPTION
**Pull request by harisriaz17**
- I implemented ResDUNet model and I am getting good results on coarse 90x17x17 Inputs. Here's one of the results for 4k train, 1k test and 5k apply events: 

resdunet_v2

This is good compared to the model running on default settings: 
unet1

- It is very easy to switch between different models by changing a few settings in database_parameters_DNN_fluctuations.yml (the new settings are commented out for now).

- The new model seems to overfit on finer 180x33x33 inputs giving bad plots and I'm working on tweaking the hyperparams for that case.

### CODE REVIEW

**Comment by ginnocen** [comment]
@saganatt. Please can you have a look at it?

**Comment by saganatt** [comment]
1. Have you compared the networks with deterministic settings? (see issues tab) Even the same configuration can give completely different results due to tf randomness.
2. Could you see and fix pylint errors? Let's keep code clean ☺

File: tpcwithdnn/dnn_optimiser.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # pylint: disable=too-many-instance-attributes, too-many-statements, too-many-arguments, fixme
 # pylint: disable=missing-module-docstring, missing-function-docstring, missing-class-docstring
 # pylint: disable=protected-access, too-many-locals
+# pylint: disable=line-too-long
```
Comment (saganatt) [inline tpcwithdnn/dnn_optimiser.py:4]: I think we should drop this line and solve the actual problem ;-)

File: tpcwithdnn/utilities_dnn.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # pylint: disable=too-many-arguments, invalid-name
 # pylint: disable=missing-module-docstring, missing-function-docstring
+# pylint: disable=unused-variable
```
Comment (saganatt) [inline tpcwithdnn/utilities_dnn.py:3]: Should it be removed, too?

File: tpcwithdnn/dnn_optimiser.py (hunk 1 above)
Comment (harisriaz17) [reply tpcwithdnn/dnn_optimiser.py]: Yep, fixed.

File: tpcwithdnn/utilities_dnn.py (hunk 2 above)
Comment (harisriaz17) [reply tpcwithdnn/utilities_dnn.py]: Yep :).

**Comment by saganatt** [comment]
Thanks, Hariz! I will look at these codes again tomorrow, but for now it looks fine to me.

**Pull request closed without merging**